index.js

```diff
@@ -52,6 +53,10 @@ module.exports = options => {
 			if (props.type) {
 				const type = props.type;
 
+				if (availableTypes.indexOf(type) === -1) {
+					throw new TypeError(`Expected "${key}" to be boolean or string, got ${type}`);
+				}
+
 				if (type === 'string') {
```

Comment:
I think with the current logic here, all non string/boolean types will not be added to the final options object. Since you're validating `type` above already, here we can just have: `push(result, type, key)`.

Reply:
As you said, `availableTypes` contains four types
```
availableTypes = ['string', 'boolean', 'number', 'object'];
```
To be able to use `push(result, type, key)`, I will need to exclude `number` and `object` for this to work.

Reply:
You mean that `minimist-options` won't support `number`? I think that's ok, we can add a note about it in readme.

Re `object`, I forgot, why did we add it if it's not supported anywhere? :D

Reply:
`object` was added to support array like object.

Reply:
I've added `number` and `object` to the final object.